src/compiler/moduleSpecifiers.ts

```diff
@@ -111,6 +112,15 @@ import {
     UserPreferences,
 } from "./_namespaces/ts";
 
+function safeJsonRead(packageJsonPath: string, readFile: (path: string) => string | undefined) {
+    try {
+        return JSON.parse(readFile(packageJsonPath)!);

```

Comment:
on one hand, this is more correct (per past discussion), on the other, it's sorta unfortunate that someone editing their package.json will just not get _any_ package.json stuff for module resolution if they temporarily have a trailing comma or something. Oh well